transaction/src/action/swap_claim.rs

```diff
@@ -19,42 +21,26 @@ impl IsAction for SwapClaim {
     }
 
     fn view_from_perspective(&self, txp: &TransactionPerspective) -> ActionView {
-        // For each note payload (output_1, output_2)
-        let note_commitment_1 = self.body.output_1.note_commitment;
-        let note_commitment_2 = self.body.output_2.note_commitment;
-        // Get payload key for note commitment of note payload
-        let payload_key_1 = txp.payload_keys.get(&note_commitment_1);
-        let payload_key_2 = txp.payload_keys.get(&note_commitment_2);
-
-        let swap_claim_view = if let (Some(payload_key_1), Some(payload_key_2)) =
-            (payload_key_1, payload_key_2)
-        {
-            // * Decrypt notes
-            let decrypted_note_1 =
-                Note::decrypt_with_payload_key(&self.body.output_1.encrypted_note, payload_key_1);
-            let decrypted_note_2 =
-                Note::decrypt_with_payload_key(&self.body.output_2.encrypted_note, payload_key_2);
+        // Get the advice notes for each output from the swap claim
+        let output_1 = txp.advice_notes.get(&self.body.output_1_commitment);
+        let output_2 = txp.advice_notes.get(&self.body.output_2_commitment);
```

Comment:
To view a `SwapClaim`'s output notes, we need to use the new advice mechanism to be able to get the openings of the rolled-up output note commitments.